### PR TITLE
Add com.valvesoftware.Steam.CompatibilityTool.Boxtron

### PR DESCRIPTION
**Pull request by gasinvein**
[Boxtron](https://github.com/dreamer/boxtron) is a Steam compatibility tool similar to Proton, but instead of running Windows games via Wine, it runs DOS games via Dosbox.

Currently full support for 3rd-party compatibility tools is added to Steam beta (opt-in in the client), but is expected to land in the release soon.

CC @dreamer @nanonyme @AsciiWolf

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

File: com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+  - name: inotify-tools
+    sources:
+      - type: archive
+        url: "https://downloads.sourceforge.net/inotify-tools/inotify-tools-3.13.tar.gz"
+        sha256: 464146acec2c118be63a6c2a090d9252fbe9689c093db85b695d0502fa229445
+
+  - name: Boxtron
+    #TODO enable linting, requires shellcheck and pylint
+    buildsystem: simple
```
Comment (nanonyme) [inline com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml]: Would it be too much effort to build Boxtron ourselves against org.freedesktop.Sdk?

**Comment by nanonyme** [comment]
Any chance having a rebase after you get it to work? The commit history in this branch seems quite weird.

File: com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml (hunk 1 above)
Comment (gasinvein) [reply com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml]: It's written in python, and is kind of built from source here.

**Comment by gasinvein** [comment]
@nanonyme I can squash and force-push, sure, but is this commit history that bad?

**Comment by gasinvein** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by LeandroStanger** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

File: com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+id: com.valvesoftware.Steam.CompatibilityTool.Boxtron
+branch: stable
+runtime: com.valvesoftware.Steam
+runtime-version: stable
+sdk: org.freedesktop.Sdk//19.08
+build-extension: true
+appstream-compose: false
+build-options:
+  prefix: /app/share/steam/compatibilitytools.d/Boxtron
```
Comment (gasinvein) [inline com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml]: @dreamer, should the install directory be `boxtron` (lowercase)?

**Comment by nanonyme** [comment]
@gasinvein the problem is you have commits here that aren't part of your work but things that are already either merged to trunk or dropped. Branches should only have commits that are part of that feature. Squash is overkill, having development effort visible through commit history is imo fine.

File: com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml (hunk 1 above)
Comment (nanonyme) [reply com.valvesoftware.Steam.CompatibilityTool.Boxtron.yml]: Fair enough. Sounds good to me then.

**Comment by nanonyme** [comment]
I only now realized this is a separate repo so I stopped caring about history tidiness. Sounds good to me! This is actually material that I would consider allowing to be auto-installed in Steam app.

**Comment by gasinvein** [comment]
@nanonyme What auto-install do you mean?

**Comment by LeandroStanger** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by fishxz** [comment]
nice.

slightly OT: are you interested to add https://github.com/GloriousEggroll/proton-ge-custom/releases ? would be nice to have this here as well, so i/we don't have to check for updates manually :)

**Comment by LeandroStanger** [comment]
>slightly OT: are you interested to add https://github.com/GloriousEggroll/proton-ge-custom/releases ? would be nice to have this here as well, so i/we don't have to check for updates manually :)

https://github.com/flathub/com.valvesoftware.Steam/issues/419

**Comment by gasinvein** [comment]
@fishxz Proton build system heavily relies on Vagrant and Docker, and getting it to build with flatpak-builder is going to be a nightmare. I've started to [work on it](https://github.com/gasinvein/flatpak-apps-staging/tree/com.valvesoftware.Steam.CompatibilityTool.Proton-i/master), but don't expect actual results anytime soon, any help appreciated.

**Comment by nanonyme** [comment]
@dreamer I'm curious, why do you think it will cause issues to run Proton linked against freedesktop-sdk?

**Comment by LeandroStanger** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by LeandroStanger** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by dreamer** [comment]
URL still returns 404… Do we need to do something to get access to these builds?

**Comment by nanonyme** [comment]
Ping @barthalion any idea about the 404?

**Comment by LeandroStanger** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by barthalion** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by barthalion** [comment]
https://github.com/flatpak/flat-manager/pull/32 will fix generating flatpakref files for non-apps refs. The underlying repo is there so you can install the build above with this file:

```
[Flatpak Ref]
Name=com.valvesoftware.Steam.CompatibilityTool.Boxtron
Branch=stable
Title=com.valvesoftware.Steam.CompatibilityTool.Boxtron test build
IsRuntime=true
Url=https://dl.flathub.org/build-repo/10559
GPGKey=mQINBFlD2sABEADsiUZUOYBg1UdDaWkEdJYkTSZD68214m8Q1fbrP5AptaUfCl8KYKFMNoAJRBXn9FbE6q6VBzghHXj/rSnA8WPnkbaEWR7xltOqzB1yHpCQ1l8xSfH5N02DMUBSRtD/rOYsBKbaJcOgW0K21sX+BecMY/AI2yADvCJEjhVKrjR9yfRX+NQEhDcbXUFRGt9ZT+TI5yT4xcwbvvTu7aFUR/dH7+wjrQ7lzoGlZGFFrQXSs2WI0WaYHWDeCwymtohXryF8lcWQkhH8UhfNJVBJFgCY8Q6UHkZG0FxMu8xnIDBMjBmSZKwKQn0nwzwM2afskZEnmNPYDI8nuNsSZBZSAw+ThhkdCZHZZRwzmjzyRuLLVFpOj3XryXwZcSefNMPDkZAuWWzPYjxS80cm2hG1WfqrG0Gl8+iX69cbQchb7gbEb0RtqNskTo9DDmO0bNKNnMbzmIJ3/rTbSahKSwtewklqSP/01o0WKZiy+n/RAkUKOFBprjJtWOZkc8SPXV/rnoS2dWsJWQZhuPPtv3tefdDiEyp7ePrfgfKxuHpZES0IZRiFI4J/nAUP5bix+srcIxOVqAam68CbAlPvWTivRUMRVbKjJiGXIOJ78wAMjqPg3QIC0GQ0EPAWwAOzzpdgbnG7TCQetaVV8rSYCuirlPYN+bJIwBtkOC9SWLoPMVZTwQARAQABtC5GbGF0aHViIFJlcG8gU2lnbmluZyBLZXkgPGZsYXRodWJAZmxhdGh1Yi5vcmc+iQJUBBMBCAA+FiEEblwF2XnHba+TwIE1QYTdTZB6fK4FAllD2sACGwMFCRLMAwAFCwkIBwIGFQgJCgsCBBYCAwECHgECF4AACgkQQYTdTZB6fK5RJQ/+Ptd4sWxaiAW91FFk7+wmYOkEe1NY2UDNJjEEz34PNP/1RoxveHDt43kYJQ23OWaPJuZAbu+fWtjRYcMBzOsMCaFcRSHFiDIC9aTp4ux/mo+IEeyarYt/oyKb5t5lta6xaAqg7rwt65jW5/aQjnS4h7eFZ+dAKta7Y/fljNrOznUp81/SMcx4QA5G2Pw0hs4Xrxg59oONOTFGBgA6FF8WQghrpR7SnEe0FSEOVsAjwQ13Cfkfa7b70omXSWp7GWfUzgBKyoWxKTqzMN3RQHjjhPJcsQnrqH5enUu4Pcb2LcMFpzimHnUgb9ft72DP5wxfzHGAWOUiUXHbAekfq5iFks8cha/RST6wkxG3Rf44Zn09aOxh1btMcGL+5xb1G0BuCQnA0fP/kDYIPwh9z22EqwRQOspIcvGeLVkFeIfubxpcMdOfQqQnZtHMCabV5Q/Rk9K1ZGc8M2hlg8gHbXMFch2xJ0Wu72eXbA/UY5MskEeBgawTQnQOK/vNm7t0AJMpWK26Qg6178UmRghmeZDj9uNRc3EI1nSbgvmGlpDmCxaAGqaGL1zW4KPW5yN25/qeqXcgCvUjZLI9PNq3Kvizp1lUrbx7heRiSoazCucvHQ1VHUzcPVLUKKTkoTP8okThnRRRsBcZ1+jI4yMWIDLOCT7IW3FePr+3xyuy5eEo9a25Ag0EWUPa7AEQALT/CmSyZ8LWlRYQZKYw417p7Z2hxqd6TjwkwM3IQ1irumkWcTZBZIbBgrSOg6CcXD2oWydCQHWi9qaxhuhEl2bJL5LskmBcMxVdQeD0LLHd8QUnbnnIby8ocvWN1alPfvJFjCUTrmD22U1ycOzRw2lIe4kiQONbOZtdWrVImQQSndjFlisitbmlWHvHm2lOOYy8+GJB7YffVV193hmnBSJffCy4bvkuLxsI+n1DhOzc7MPV3z6HGk4HiEcF0yyt9tCYhpsxHFdBoq2h771HfAcS0s98EVAqYMFnf9em+4cnYpdI6mhIfS1FQiKl6DBAYA8tT3ggla00DurPo0JwX/zN+PaO5h/6O9aCZwV7G6rbkgMuqMergXaf8oP38gr0z+MqWnkfM63Bodq68GP4l4hd02BoFBbDf38TMuGQB14+twJMdfbAxo2MbgluvQgfwHfZ2ca6gyEY+9s/YD1gugLjV+S6CB51WkFNe1z4tAPgJZNxUcKCbeaHNbthl8Hks/pY9RCEseX/EdfzF18epbSjJMPh4DPQXbUoFwmyuYcoBOPmvZHNl9hK7B/1RP8w1ZrXk8qdupC0SNbafX7270B7lMMVImzZetGsM9ypXJ6llhp3FwW09iseNyGJGPsr/dvTMGDXqOPfU/9SAS1LSTY4K9PbRtdrBE318YX8mIk5ABEBAAGJBHIEGAEIACYWIQRuXAXZecdtr5PAgTVBhN1NkHp8rgUCWUPa7AIbAgUJEswDAAJACRBBhN1NkHp8rsF0IAQZAQgAHRYhBFSmzd2JGfsgQgDYrFYnAunj7X7oBQJZQ9rsAAoJEFYnAunj7X7oR6AP/0KYmiAFeqx14Z43/6s2gt3VhxlSd8bmcVV7oJFbMhdHBIeWBp2BvsUf00I0Zl14ZkwCKfLwbbORC2eIxvzJ+QWjGfPhDmS4XUSmhlXxWnYEveSek5Tde+fmu6lqKM8CHg5BNx4GWIX/vdLi1wWJZyhrUwwICAxkuhKxuP2Z1An48930eslTD2GGcjByc27+9cIZjHKa07I/aLffo04V+oMT9/tgzoquzgpVV4jwekADo2MJjhkkPveSNI420bgT+Q7Fi1l0X1aFUniBvQMsaBa27PngWm6xE2ZYvh7nWCdd5g0c0eLIHxWwzV1lZ4Ryx4ITO/VL25ItECcjhTRdYa64sA62MYSaB0x3eR+SihpgP3wSNPFu3MJo6FKTFdi4CBAEmpWHFW7FcRmd+cQXeFrHLN3iNVWryy0HK/CUEJmiZEmpNiXecl4vPIIuyF0zgSCztQtKoMr+injpmQGC/rF/ELBVZTUSLNB350S0Ztvw0FKWDAJSxFmoxt3xycqvvt47rxTrhi78nkk6jATKGyvP55sO+K7Q7Wh0DXA69hvPrYW2eu8jGCdVGxi6HX7L1qcfEd0378S71dZ3g9o6KKl1OsDWWQ6MJ6FGBZedl/ibRfs8p5+sbCX3lQSjEFy3rx6n0rUrXx8U2qb+RCLzJlmC5MNBOTDJwHPcX6gKsUcXZrEQALmRHoo3SrewO41RCr+5nUlqiqV3AohBMhnQbGzyHf2+drutIaoh7Rj80XRh2bkkuPLwlNPf+bTXwNVGse4bej7B3oV6Ae1N7lTNVF4Qh+1OowtGjmfJPWo0z1s6HFJVxoIof9z58Msvgao0zrKGqaMWaNQ6LUeC9g9Aj/9Uqjbo8X54aLiYs8Z1WNc06jKP+gv8AWLtv6CR+l2kLez1YMDucjm7v6iuCMVAmZdmxhg5I/X2+OM3vBsqPDdQpr2TPDLX3rCrSBiS0gOQ6DwN5N5QeTkxmY/7QO8bgLo/Wzu1iilH4vMKW6LBKCaRx5UEJxKpL4wkgITsYKneIt3NTHo5EOuaYk+y2+Dvt6EQFiuMsdbfUjs3seIHsghX/cbPJa4YUqZAL8C4OtVHaijwGo0ymt9MWvS9yNKMyT0JhN2/BdeOVWrHk7wXXJn/ZjpXilicXKPx4udCF76meE+6N2u/T+RYZ7fP1QMEtNZNmYDOfA6sViuPDfQSHLNbauJBo/n1sRYAsL5mcG22UDchJrlKvmK3EOADCQg+myrm8006LltubNB4wWNzHDJ0Ls2JGzQZCd/xGyVmUiidCBUrD537WdknOYE4FD7P0cHaM9brKJ/M8LkEH0zUlo73bY4XagbnCqve6PvQb5G2Z55qhWphd6f4B6DGed86zJEa/RhS
RuntimeRepo=https://dl.flathub.org/repo/flathub.flatpakrepo
```

**Comment by barthalion** [comment]
@nanonyme Are you good with it? I'm mostly waiting for your ack with merge.

**Comment by nanonyme** [comment]
Oh, sorry. I was assuming this was pending comment from @dreamer. Yes, I think this is good.

**Comment by barthalion** [comment]
bot, build com.valvesoftware.Steam.CompatibilityTool.Boxtron

**Comment by barthalion** [comment]
@dreamer URL above finally works if you'd like to test it.

**Comment by gasinvein** [comment]
@barthalion Seems like @dreamer (the developer) did'n receive invitation to collaborate on the repo. Can you please re-send it?

**Comment by bilelmoussaoui** [comment]
@gasinvein Done

**Pull request closed without merging**